### PR TITLE
Feat/content safety

### PR DESCRIPTION
**Pull request by Pouyanpi**
This PR introduces the content safety module, enabling users to utilize various models through a "content safety check input/output" flow. 

Key enhancements include:

- Parameter support in input/output rails flow definitions.
- The addition of the `is_content_safe` output parser, which is shared between the `content_safety` and `self_check` modules. This new implementation maintains backward compatibility.
- The introduction of a `max_tokens` field in TaskPrompt.

TODO and Remarks:
- [x] Make shieldgemma example work, currently shieldgemma is not behaving correctly.
- [ ] _MAX_TOKENS or default value of max_tokens in TaskPrompt
- [ ] check functionality within colang 2
- [x] Documentation

### CODE REVIEW

File: nemoguardrails/rails/llm/config.py [hunk 1]
```diff
@@ -163,6 +164,11 @@ class TaskPrompt(BaseModel):
         description="If specified, will be configure stop tokens for models that support this.",
     )
 
+    max_tokens: Optional[int] = Field(
+        default=None,
```
Comment (Pouyanpi) [inline nemoguardrails/rails/llm/config.py:168]: is `None` Ok?

File: nemoguardrails/library/content_safety/actions.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+
+@action()
+async def content_safety_check_output(
+    llms: Dict[str, BaseLLM],
+    llm_task_manager: LLMTaskManager,
+    model_name: Optional[str] = None,
+    context: Optional[dict] = None,
+) -> dict:
+    _MAX_TOKENS = 3
```
Comment (Pouyanpi) [inline nemoguardrails/library/content_safety/actions.py:100]: `max_tokens` of None is not supported by some providers.

File: nemoguardrails/library/content_safety/actions.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        },
+    )
+    stop = llm_task_manager.get_stop_tokens(task=task)
+    max_tokens = llm_task_manager.get_max_tokens(task=task)
+
+    max_tokens = max_tokens or _MAX_TOKENS
+
+    llm_call_info_var.set(LLMCallInfo(task=task))
+
+    with llm_params(llm, temperature=1e-20, max_tokens=max_tokens):
```
Comment (Pouyanpi) [inline nemoguardrails/library/content_safety/actions.py:142]: gemmashield or ChatNVIDIA has `gt = 0.0` constraint on `max_tokens` field and not `ge`.

Comment (drazvan) [reply nemoguardrails/library/content_safety/actions.py:142]: Ack.

**Pull request merged**